lib/prestashopper/api.rb

```diff
@@ -22,33 +22,28 @@ def initialize(url, key)
     # List resources that the API key can access
     # @return [Array<Symbol>] list of resources the API can access
     def resources
-      xml = @resources_res.get.body
-      xml_doc = Nokogiri::XML xml
-      nodes = xml_doc.xpath '/prestashop/api/*'
-      resources_list = []
-      nodes.each{|n| resources_list << n.name.to_sym}
-      return resources_list
+      @resources ||= Nokogiri::XML(@resources_res.get.body).xpath('/prestashop/api/*').collect { |resource| resource.name.to_sym }
```

Comment:
This was mostly merged into one line to make it more clear that no `get` or parsing is done if `@resoures` is already set.